SimpleRabbit.NetCore/Service/IMessageHandler.cs

```diff
@@ -1,50 +1,19 @@
 ﻿namespace SimpleRabbit.NetCore
 {
-    /// <summary>
-    /// An Abstraction of a component that can consume messages
-    /// </summary>
-    /// <remarks>
-    /// <para>
-    /// A basic example of a subscriber is as follows
-    /// </para>
-    /// <example>
-    /// <code>
-    /// public class MessageProcessor : IMessageHandler
-    /// {
-    ///     public bool CanProcess(string tag)
-    ///     {
-    ///         return true;
-    ///     }
-    /// 
-    ///     public bool Process(BasicMessage message)
-    ///     {
-    ///         Console.WriteLine(message.Body);
-    ///         return true;
-    ///     }
-    /// }
-    /// </code>
-    /// 
-    /// </example>
-    /// </remarks> 
     public interface IMessageHandler
     {
         /// <summary>
-        /// Determine whether this handler will handle this message
+        /// Determine whether this handler will handle this message.
         /// </summary>
-        /// <param name="tag"> the <see cref="QueueConfiguration.ConsumerTag"/> determined by which queue a message came from</param>
-        /// <returns>boolean true if this handler will take this message, false otherwise</returns>
+        /// <param name="tag"> the <see cref="QueueConfiguration.ConsumerTag"/> determined by which queue a message came from.</param>
+        /// <returns>boolean true if this handler will take this message, false otherwise.</returns>
         bool CanProcess(string tag);
+
         /// <summary>
-        /// Consume a message 
+        /// Consume and process a message.
         /// </summary>
-        /// <param name="message">A message containing headers and the raw data</param>
-        /// <returns>flag indicating if this message should be acknowledged</returns>
-        /// <remarks>
-        /// <para>
-        /// return true if you want to ack immediately. 
-        /// false allows for delegation of the acknowledgement, either by dispatching to a thread or acknowledging later.
-        /// </para>
-        /// </remarks>
-        bool Process(BasicMessage message);
+        /// <param name="message">A message containing headers and the raw data.</param>
+        /// <returns><see cref="Acknowledgement"/> enum denoting how the message should be acknowledged.</returns>
+        Acknowledgement Process(BasicMessage message);
```

Comment:
With the breaking change introduced, it's a good time to expose `Task<Acknowledgement> Process(BasicMessage message);` as async eventing is supported by RabbitClient

Reply:
Do you have any thoughts on how to handle this? Are we deprecating the synchronous version of Process?